tests/_internal/test_resolve_target.py

```diff
@@ -78,15 +70,19 @@ def test_will_resolve_targets_using_online_id_when_available(
 
     @mock.patch("mbed_devices._internal.resolve_target.read_product_code")
     @mock.patch("mbed_devices._internal.resolve_target.read_online_id")
-    def test_raises_when_no_information_found_on_candidate(
-        self, read_product_code, read_online_id, _get_all_htm_files_contents
+    @mock.patch("mbed_devices._internal.resolve_target.get_target_by_product_code")
+    def test_will_resolve_targets_by_product_code_from_serial_number_if_htm_file_has_no_information(
```

Comment:
The name of this test is a bit too long for me, how about "test_falls_back_to_serial_number_resolver" and put any additional information in a docstring.